File: frontend/src/types/feedback.ts
export interface Feedback {
  id: number
  ticket_id: number
  ai_response_id: number
  rating: number
  is_helpful: boolean | null
  comment: string | null
  created_at: string
}

export interface FeedbackCreate {
  ai_response_id: number
  rating: number
  is_helpful?: boolean
  comment?: string
}

File: backend/app/schemas/feedback.py
from datetime import datetime

from pydantic import BaseModel, Field


class FeedbackCreate(BaseModel):
    ai_response_id: int
    rating: int = Field(..., ge=1, le=5)
    is_helpful: bool | None = None
    comment: str | None = None


class FeedbackUpdate(BaseModel):
    rating: int | None = Field(None, ge=1, le=5)
    is_helpful: bool | None = None
    comment: str | None = None


class FeedbackResponse(BaseModel):
    id: int
    ticket_id: int
    ai_response_id: int
    rating: int
    is_helpful: bool | None
    comment: str | None
    created_at: datetime

    model_config = {"from_attributes": True}
